src/Program.cs

```diff
@@ -26,21 +28,51 @@ static void Main(string[] args)
                 if (File.Exists(inputPath))
                 {
                     CommandLineUtils.CreateOutputDirectory(outputPath);
-                    HtmlProcessor.ProcessFile(inputPath, outputPath);
+
+                    // Check file extension to decide whether to use MdProcessor or HtmlProcessor
+                    string fileExtension = Path.GetExtension(inputPath);
+                    if (fileExtension == ".md")
+                    {
+                        MdProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else if (fileExtension == ".txt")
+                    {
+                        HtmlProcessor.ProcessFile(inputPath, outputPath);
+                    }
+                    else
+                    {
+                        CommandLineUtils.Logger($"Unsupported file format for input path: {inputPath}");
+                    }
                 }
                 else if (Directory.Exists(inputPath))
                 {
-                    string[] files = Directory.GetFiles(inputPath, "*.txt");
+                    // Get all files in the directory and save them to the files array
+                    string[] files = Directory.GetFiles(inputPath, "*.*"); 
+
                     if (files.Length == 0)
                     {
-                        CommandLineUtils.Logger($"No txt files found in directory {inputPath}");
+                        CommandLineUtils.Logger($"No files found in directory {inputPath}");
                         return;
                     }
 
                     CommandLineUtils.CreateOutputDirectory(outputPath);
-                    foreach (string file in files)
+
+                    foreach (string file in files) // Go through the files array and convert ones that end with .md or .txt
                     {
-                        HtmlProcessor.ProcessFile(file, outputPath);
+                        // Check file extension to decide whether to use MdProcessor or HtmlProcessor.
+                        string fileExtension = Path.GetExtension(file);
+                        if (fileExtension == ".md")
+                        {
+                            MdProcessor.ProcessFile(file, outputPath);
+                        }
+                        else if (fileExtension == ".txt")
+                        {
+                            HtmlProcessor.ProcessFile(file, outputPath);
+                        }
+                        else
+                        {
+                            CommandLineUtils.Logger($"Unsupported file format for file: {file}");
```

Comment:
Unnecessary because before we reach this point of the code, we already only retrieve .md or .txt files